pkg/client/unversioned/discovery_client.go

```diff
@@ -154,7 +140,11 @@ func (d *DiscoveryClient) ServerResources() (map[string]*unversioned.APIResource
 
 func setDiscoveryDefaults(config *Config) error {
 	config.Prefix = ""
-	config.Version = ""
+	// "unversioned" is a bogus value to pass the check in RESTClientFor.
+	// TODO: make a RESTClient constructor that accepts empty version.
+	config.Version = "unversioned"
```

Comment:
Definitely not "unversioned", that's very confusing.  Also, doing this builds a `RESTClient` that has an invalid `baseURL` which disturbs me.

If you need to be able specify an empty version, I would rather see you refactor the code you're calling.  Even if it means another method to do it cleanly/easily.